zh_CN/rule/rule-engine_buildin_function.md

```diff
@@ -0,0 +1,304 @@
+pad('abc', 5, 'trailing', '*') = 'abc**'
+pad('abc', 5, 'trailing', '*#') = 'abc*#*#'
+pad('abc', 5, 'both', '*') = '*abc*'
+pad('abc', 5, 'both', '*#') = '*#abc*#'
+pad('abc', 5, 'leading', '*') = '**abc'
+pad('abc', 5, 'leading', '*#') = '*#*#abc'
+
+replace('ababef', 'ab', 'cd') = 'cdcdef'
+replace('ababef', 'ab', 'cd', 'all') = 'cdcdef'
+replace('ababef', 'ab', 'cd', 'trailing') = 'abcdef'
+replace('ababef', 'ab', 'cd', 'leading') = 'cdabef'
+
+regex_match('abc123', '[a-zA-Z1-9]*') = true
+
+regex_replace('ab1cd3ef', '[1-9]', '[&]') = 'ab[1]cd[3]ef'
+regex_replace('ccefacef', 'c+', ':') = ':efa:ef'
+
+ascii('a') = 97
+
+find('eeabcabcee', 'abc') = 'abcabcee'
+find('eeabcabcee', 'abc', 'leading') = 'abcabcee'
+find('eeabcabcee', 'abc', 'trailing') = 'abcee'
+```
+
+
+## Map 函数
+
+|函数名|函数作用|参数|返回值|
+|--- |--- |--- |--- |
+|map_get|取 Map 中某个 Key 的值，如果没有则返回空值|1. Key <br />2. Map|Map 中某个 Key 的值。支持嵌套的 Key，比如 "a.b.c"|
+|map_get|取 Map 中某个 Key 的值，如果没有则返回指定默认值|1. Key <br />2. Map <br />3. Default Value|Map 中某个 Key 的值。支持嵌套的 Key，比如 "a.b.c"|
+|map_put|向 Map 中插入值|1. Key <br />2. Value <br />3. Map|插入后的 Map。支持嵌套的 Key，比如 "a.b.c"|
+
+```erlang
+map_get('a', json_decode( '{ "a" : 1 }' )) = 1
+map_get('b', json_decode( '{ "a" : 1 }' ), 2) = 2
+map_get('a', map_put('a', 2, json_decode( '{ "a" : 1 }' ))) = 2
+```
+
+
+## 数组函数
+
+|函数名|函数作用|参数|返回值|
+|--- |--- |--- |--- |
+|nth|取第 n 个元素，下标从 1 开始|1. 起始位置 <br /> 2. 原数组|第 n 个元素|
+|length|获取数组的长度|1. 原数组|数组长度|
+|sublist|取从第一个元素开始、长度为 len 的子数组。下标从 1 开始|1. 长度 len <br />2. 原数组|子数组|
+|sublist|取从第 n 个元素开始、长度为 len 的子数组。下标从 1 开始|1. 起始位置 n <br />2. 长度 len <br />3. 原数组|子数组|
+|first|取第 1 个元素。下标从 1 开始|1. 原数组|第 1 个元素|
+|last|取最后一个元素。|1. 原数组|最后一个元素|
+|contains|判断数据是否在数组里面|1. 数据 <br />2. 原数组|Boolean 值|
+
+```erlang
+nth(2, [1,2,3,4]) = 2
+length([1,2,3,4]) = 4
+sublist(3, [1,2,3,4]) = [1,2,3,4]
+sublist(1,2,[1,2,3,4]) = [1, 2]
+first([1,2,3,4]) = 1
+last([1,2,3,4]) = 4
+contains(2, [1,2,3,4]) = true
+```
+
+## 哈希函数
+
+| 函数名 | 函数功能| 参数| 返回值 |
+|-------| ------ | -- | ----- |
+| md5	| 求 MD5 值 | 数据 | MD5 值 |
+| sha	| 求 SHA 值 | 数据 | SHA 值 |
+| sha256 | 求 SHA256 值 | 数据 | SHA256 值 |
+
+```erlang
+md5('some val') = '1b68352b3e9c2de52ffd322e30bffcc4'
+sha('some val') = 'f85ba28ff5ea84a0cbfa118319acb0c5e58ee2b9'
+sha256('some val') = '67f97635d8a0e064f60ba6e8846a0ac0be664f18f0c1dc6445cd3542d2b71993'
+```
+
+## 比特操作函数
+
+###`subbits`
+| 函数名 | 函数功能| 参数| 返回值 |
+| ----- | ------ | -- | ----- |
+|subbits| 从二进制数据的起始位置获取指定长度的比特位, 然后转换为无符号整型 (大端). | 1. 二进制数据 <br />2. 要获取的长度(bits) | 无|subbits符号整数 |
+|subbits| 从二进制数据的指定下标位置获取指定长度的比特位, 然后转换为无符号整型 (大端). 下标是从 1 开始的 | 1. 二进制数据 <br />2. 起|subbits始位置的下标 <br />3. 要获取的长度(bits) | 无符号整数 |
+|subbits| 从二进制数据的指定下标位置获取指定长度的比特位, 然后按照给定的参数转换为想要的数据类型. 下标是从 1 开始的. | 1. 二进制数据 <br />2. 起始位置的下标 <br />3. 要获取的长度(bits) <br />4. 数据类型，可选值：'integer', 'float', 'bits' <br />5. 符号类型, 只对整型数据有效, 可选值：'unsigned', 'signed', <br />6. 大端还是小端, 只对整型数据有效, 可选值：'big', 'little' | 获取到的数据 |
+
+```erlang
+subbits('aaa', 10) = 389
```

Comment:
We should use 8 (or Multiple of 8) as the `length` parameter, as the binary to be parse is a string. Like:

```
subbits('aaa', 8) = 97
```